src/Evolution/Systems/ScalarWave/Equations.hpp

```diff
@@ -97,21 +98,123 @@ struct ComputeNormalDotFluxes {
+ *      (n_i v^{-})_{\mathrm{int}} + (n_i v^{+})_{\mathrm{ext}}\right)
+ * \f}
+ *
+ * where \f$G\f$ is the interface normal dotted with numerical flux, the
+ * first terms on the RHS for \f$G(\Pi)\f$ and \f$G(\Phi_i)\f$ are the fluxes
+ * dotted with the interface normal (computed in
+ * ScalarWave::ComputeNormalDotFluxes), and \f$v^{\pm}\f$ are outgoing and
+ * incoming characteristic fields of the system (see characteristic_fields() for
+ * their definition). The penalty factor is chosen to be \f$p=1\f$.
+ */
+template <size_t Dim>
+struct PenaltyFlux {
+ private:
+  struct NormalTimesVPlus {
+    using type = tnsr::i<DataVector, Dim, Frame::Inertial>;
+    static std::string name() noexcept { return "NormalTimesVPlus"; }
+  };
+  struct NormalTimesVMinus {
+    using type = tnsr::i<DataVector, Dim, Frame::Inertial>;
+    static std::string name() noexcept { return "NormalTimesVMinus"; }
+  };
+
+ public:
+  using options = tmpl::list<>;
+  static constexpr OptionString help = {
+      "Computes the penalty flux for a scalar wave system. It requires no "
+      "options."};
+  static std::string name() noexcept { return "Penalty"; }
+
+  // clang-tidy: non-const reference
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+
+  // This is the data needed to compute the numerical flux.
+  // `dg::SendBoundaryFluxes` calls `package_data` to store these tags in a
+  // Variables. Local and remote values of this data are then combined inside
+  // `operator()`.
+  using package_tags =
+      tmpl::list<::Tags::NormalDotFlux<Pi>, ::Tags::NormalDotFlux<Phi<Dim>>,
+                 Tags::VPlus, Tags::VMinus, NormalTimesVPlus,
+                 NormalTimesVMinus>;
+
+  // These tags on the interface of the element are passed to
+  // `package_data` to provide the data needed to compute the numerical fluxes.
+  using argument_tags =
+      tmpl::list<::Tags::NormalDotFlux<Pi>, ::Tags::NormalDotFlux<Phi<Dim>>,
+                 Tags::VPlus, Tags::VMinus,
+                 ::Tags::Normalized<::Tags::UnnormalizedFaceNormal<Dim>>>;
+
+  // pseudo-interface: used internally by Algorithm infrastructure, not
+  // user-level code
+  // Following the not-null pointer to packaged_data, this function expects as
+  // arguments the databox types of the `argument_tags`.
+  void package_data(
+      gsl::not_null<Variables<package_tags>*> packaged_data,
+      const Scalar<DataVector>& normal_dot_flux_pi,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& normal_dot_flux_phi,
+      const Scalar<DataVector>& v_plus, const Scalar<DataVector>& v_minus,
+      const tnsr::i<DataVector, Dim, Frame::Inertial>& interface_unit_normal)
+      const noexcept;
+
+  // pseudo-interface: used internally by Algorithm infrastructure, not
+  // user-level code
+  // The arguments are first the system::variables_tag::tags_list wrapped in
+  // Tags::NormalDotNumericalFLux as not-null pointers to write the results
```

Comment:
These comments come from the Poisson code (which has been replaced with a generic flux since then). They were more for myself to remember how the functions are called. I think you can just remove them. As commented below, I intend to add a protocol to define and document the interface for numerical fluxes.

Reply:
rephrased to read better:

```
  // The first three arguments are pointers to Tags::NormalDotNumericalFlux<...>
  // for each variable in the system, then the package_tags on the interior side
  // of the mortar followed by the package_tags on the exterior side.
```